### PR TITLE
Add Scripts for podman-based development

### PR DESCRIPTION
**Pull request by criztovyl**
#### What this PR does / why we need it:

During restructuring #517 I extracted these changes that could be added as a dedicate PR.

#### Which issue(s) this PR fixes:

Containerized, Podman-based, kapp development.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional Notes for your reviewer:

The other `hack` scripts are not documented, so I have not added docs for the new scripts (yet?). 

But people familiar with Bash and Podman should be able to understand the scripts.

##### Review Checklist:

(Does not seem applicable.)

- [x]  Follows the [developer guidelines](https://carvel.dev/shared/docs/latest/development_guidelines/)
- [x] Relevant tests are added or updated
- [ ] Relevant docs in this repo added or updated
- [x] Relevant carvel.dev docs added or updated in a separate PR and there's
  a link to that PR
- [x] Code is at least as readable and maintainable as it was before this
  change

#### Additional documentation e.g., Proposal, usage docs, etc.:

```docs
None
```

### CODE REVIEW

File: hack/.gitignore [hunk 1]
```diff
@@ -0,0 +1,2 @@
+.bash_history
+.dlv
```
Comment (praveenrewar) [inline hack/.gitignore:2]: Should we add these in the `.gitignore` present in the root?

File: hack/build-podman.sh [hunk 2]
```diff
@@ -0,0 +1,5 @@
+#!/usr/bin/env bash
+
+cd $(dirname ${BASH_SOURCE[0]})
+
+./podman.sh build
```
Comment (praveenrewar) [inline hack/build-podman.sh:5]: Do we need to run the podman.sh script from the hack directory for it to work? If so, can we make some changes to be able to run it from the root itself (like other scripts are doing, test-all.sh for example). It seems like we are adding one extra layer of complexity in build-podman.sh for cd'ing to the hack directory.

File: hack/podman.sh [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+gobuild_vol=${KAPP_GOBUILD_VOL:-gobuild}
+
+# SYS_PTRACE for dlv
+# host network for minikube
+podman run --rm \
+    -v $gopath_vol:/go \
+    -v $gobuild_vol:/root/.cache/go-build \
+    -v $HOME/.kube:/root/.kube \
+    -v $HOME/.minikube:$HOME/.minikube \
+    -v $(realpath $PWD/..):/mnt -w /mnt \
```
Comment (praveenrewar) [inline hack/podman.sh:31]: Maybe add a comment that realpath would require `brew install coreutils` on OSX?

**Review by praveenrewar — COMMENTED** [review]
When I run the script on my local, I get this error `Error: statfs /Users/prewar/Workspace/carvel-kapp: no such file or directory`. I am not sure what I might be doing wrong. I haven't gone through the script in detail but will go through it in some time.

File: hack/.gitignore (hunk 1 above)
Comment (criztovyl) [reply hack/.gitignore:2]: You mean instead of here? I can change it if you prefer.

File: hack/build-podman.sh (hunk 2 above)
Comment (criztovyl) [reply hack/build-podman.sh:5]: I'll check that.

File: hack/podman.sh (hunk 3 above)
Comment (criztovyl) [reply hack/podman.sh:31]: My podman scripts most likely are generally only compatible with Linux, not sure how the PWD mount and host network will work on OSX where Podman runs in a VM.

**Comment by criztovyl** [comment]
Thinking it over, the kapp repo might not be the best place to add these scripts, in theory the are generic for podman-/container-based go development.

Maybe I'll rather put these to a personal repo (so, not add them here at all), what do you think?

**Comment by praveenrewar** [comment]
> in theory the are generic for podman-/container-based go development.

Yep!

> Maybe I'll rather put these to a personal repo (so, not add them here at all), what do you think?

Sounds good to me :)

**Pull request closed without merging**